### PR TITLE
Add OpenSamlAssertingPartyDetails

### PR DESCRIPTION
**Pull request by jzheaux**
Closes gh-10781

### CODE REVIEW

**Comment by jzheaux** [comment]
@OrangeDog, does this meet the use case you identified in [this comment](https://github.com/spring-projects/spring-security/issues/10551#issue-1063762053):

> `RelyingPartyRegistrations.fromMetadataLocation` only results in a single `RelyingPartyRegistration`, and the full entity descriptor does not appear to be available from that - e.g. the `<EntityAttributes>` and `<Extensions>`.

**Comment by OrangeDog** [comment]
I assume it does. You guys tested it, right?

It will still need verification of the `EntitiesDescriptor` signature, especially when SSL is not available.
And ideally the ability to see and use the response's cache details, otherwise both users and the federation host get annoyed with you.

Let me go put some of my code examples on my top-level issue and you can hopefully put together an equivalent example for Spring 5/6.

File: saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/OpenSamlMetadataAssertingPartyDetailsConverter.java [hunk 1]
```diff
@@ -216,4 +223,65 @@ private Document document(InputStream inputStream) {
 		return new ArrayList<>();
 	}
 
+	public static final class AssertingPartyDetails extends RelyingPartyRegistration.AssertingPartyDetails {
```
Comment (marcusdacoregio) [inline saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/OpenSamlMetadataAssertingPartyDetailsConverter.java:226]: I am wondering why you have decided to have an inner class instead `OpenSamlAssertingPartyDetails`?

File: saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/OpenSamlMetadataAssertingPartyDetailsConverter.java [hunk 2]
```diff
@@ -216,4 +223,65 @@ private Document document(InputStream inputStream) {
...
+		public EntityDescriptor getEntityDescriptor() {
+			return this.descriptor;
+		}
+
+		/**
+		 * Use this {@link EntityDescriptor} to begin building an
+		 * {@link org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistration.AssertingPartyDetails}
+		 * @param entity the {@link EntityDescriptor} to use
+		 * @return the
+		 * {@link org.springframework.security.saml2.provider.service.registration.OpenSamlAssertingPartyDetails.Builder}
```
Comment (marcusdacoregio) [inline saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/OpenSamlMetadataAssertingPartyDetailsConverter.java:253]:
```suggestion
		 * {@link org.springframework.security.saml2.provider.service.registration.OpenSamlMetadataAssertingPartyDetailsConverter.AssertingPartyDetails.Builder}
```

File: saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/OpenSamlMetadataAssertingPartyDetailsConverter.java [hunk 3]
```diff
@@ -216,4 +223,65 @@ private Document document(InputStream inputStream) {
...
+
+			private Builder(EntityDescriptor descriptor) {
+				this.descriptor = descriptor;
+			}
+
+			/**
+			 * Build an
+			 * {@link org.springframework.security.saml2.provider.service.registration.OpenSamlAssertingPartyDetails}
+			 * @return
+			 */
```
Comment (marcusdacoregio) [inline saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/OpenSamlMetadataAssertingPartyDetailsConverter.java:277]:
```suggestion
			/**
			 * Build an
			 * {@link org.springframework.security.saml2.provider.service.registration.OpenSamlMetadataAssertingPartyDetailsConverter.AssertingPartyDetails}
			 * @return the {@link org.springframework.security.saml2.provider.service.registration.OpenSamlMetadataAssertingPartyDetailsConverter.AssertingPartyDetails}
			 */
```

File: saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java [hunk 4]
```diff
@@ -422,6 +422,21 @@ public static Builder withRegistrationId(String registrationId) {
 		return new Builder(registrationId);
 	}
 
+	public static Builder withAssertingPartyDetails(AssertingPartyDetails assertingPartyDetails) {
```
Comment (marcusdacoregio) [inline saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java:425]: Please update the copyright header of this class

**Review by marcusdacoregio — CHANGES_REQUESTED** [review]
Great @jzheaux. I've left some feedback inline.

**Review by marcusdacoregio — APPROVED** [review]

**Pull request merged**